src/west/commands/project.py

```diff
@@ -593,6 +629,20 @@ def _create_branch(project, branch):
              .format(branch))
 
 
+def _current_branch(project):
+    # Determine if project is currently on a branch
+    if not _cloned(project):
```

Comment:
Per discussion above, I think this should error out in this case.

Reply:
Current branch is only used for determine if HEAD of project is tracking a branch, so that we can inform user that he is now in detached head, and how he can rebase his branch onto the detached head.

So I don't see the reason why this should error out.
(mayby another function name ? as it sounds like you had expected another behavior based on the name)